src/main.rs

```diff
@@ -105,33 +107,81 @@ fn read_sns_canister_ids(file_path: Option<String>) -> AnyhowResult<Option<SnsCa
 
     let path = PathBuf::from(file_path);
     let file = File::open(path).context("Could not open the SNS Canister Ids file")?;
-    let ids: HashMap<String, String> =
+    let ids: HashMap<String, Value> =
         serde_json::from_reader(file).context("Could not parse the SNS Canister Ids file")?;
 
     let governance_canister_id = parse_canister_id("governance_canister_id", &ids)?;
     let ledger_canister_id = parse_canister_id("ledger_canister_id", &ids)?;
     let root_canister_id = parse_canister_id("root_canister_id", &ids)?;
 
+    let dapp_canister_id_list = match parse_dapp_canister_id_list("dapp_canister_id_list", &ids) {
+        Ok(v) => v,
+        Err(err) => {
+            eprintln!("{}", err);
```

Comment:
If parsing the canister_ids file fails, I think this should return an anyhow! error instead of continuing execution.